### PR TITLE
Add support for cookies to HttpClient, and support for getting headers from HttpResponse

### PR DESCRIPTION
**Pull request by randyv2zhurosoft**
Add support for cookies to HttpClient. This relies on FileUtils being able to access a file to store the cookies themselves between instances of CURL.

This also adds support for getting headers from HttpResponse, in case someone wants to read those in addition to response data (e.g. to handle cookies manually).

### CODE REVIEW

**Comment by dumganhar** [comment]
Does it need to save cookie? Why do we need to use cookie in games? Could you give me a hint?

File: extensions/network/HttpClient.cpp [hunk 1]
```diff
@@ -68,6 +68,7 @@
 
 typedef size_t (*write_callback)(void *ptr, size_t size, size_t nmemb, void *stream);
 
+static const char *cookieFilename = (CCFileUtils::sharedFileUtils()->getWritablePath() + "cookieFile.txt").c_str();
```
Comment (dumganhar) [inline extensions/network/HttpClient.cpp:71]: Should we write hard code of the path of cookie file?
I think providing a method to set the path is more elegant.

**Comment by hzd822** [comment]
great! cookie is need for some 3rd  channel in android.

**Comment by randyv2zhurosoft** [comment]
Cookies are often used in games which have a server login component (e.g. to store a session cookie).

The path of the cookie file can be an option, though typically it seems people don't care what the path is and just want cookies to work.

**Comment by randyv2zhurosoft** [comment]
It looks like the latest version of CCHttpRequest.cpp has changed, so I'll submit a new pull request.

**Pull request closed without merging**